### PR TITLE
Improve search attribute extraction

### PR DESCRIPTION
**Pull request by xNevo**
@stefanmcshane  As mentioned in #319, here's my PR for improving the part of the attribute extraction after an LDAP search result has been returned by the server. 

I did [some benchmarks](https://gist.github.com/xNevo/e4faf1a90bb610e59ec622e3e6e62558), as requested. The differences become more significant the more attributes are parsed. Regardless of the performance: I think this code should be moved into it's own fucnction anyway for better readability.

### CODE REVIEW

**Comment by xNevo** [comment]
May someone please look into this?

**Comment by xNevo** [comment]
@johnweldon  Is there a chance that this PR will be reviewed/merged?

**Comment by johnweldon** [comment]
> @johnweldon Is there a chance that this PR will be reviewed/merged?

😞 I'm sorry for neglecting this. I'll review shortly

File: search.go [hunk 1]
```diff
@@ -3,10 +3,9 @@ package ldap
 import (
 	"errors"
 	"fmt"
+	ber "github.com/go-asn1-ber/asn1-ber"
```
Comment (johnweldon) [inline search.go:6]: Minor nit, keep third party imports as a second import group, rather than mixed in with the stdlib imports.

File: search.go [hunk 2]
```diff
@@ -366,7 +365,8 @@ func (l *Conn) Search(searchRequest *SearchRequest) (*SearchResult, error) {
 	result := &SearchResult{
 		Entries:   make([]*Entry, 0),
 		Referrals: make([]string, 0),
-		Controls:  make([]Control, 0)}
+		Controls:  make([]Control, 0),
+	}
```
Comment (johnweldon) [inline search.go:369]: I don't mind trivial whitespace changes like this, but I'd prefer if we keep them consistent between the root version and the `v3/` version

**Review by johnweldon — CHANGES_REQUESTED** [review]
LGTM; if you could address the minor whitespace issues, and rebase, I'll get this merged.

**Review by johnweldon — APPROVED** [review]

**Pull request merged**